### DISCUSSION TITLE
Pause after mistake to understand the reason

### DISCUSSION THREAD

**Discussion by iandunn** [monkeytypegame/monkeytype#2622] [Ideas]
_This is different from #402, #2021, #2073._ It'd be off by default, and would not count towards leaderboards, challenges, etc.

----

I'm trying to work on my accuracy, but it's hard to stop and think about what led to a mistake when I'm in the middle of a test, since that would distort my stats. If I do stop and think about it, though, that often helps me understand what happened so that I can prevent it in the future.

I think it'd be helpful to have a new "mode" where the app would automatically pause after every mistake, so that the typer can reflect on what they did that led to it. It would be a kind of practice mode where you can take advantage of the word lists and other features Monkeytype has.

Once the typer is ready, they can hit `escape` to close the modal and start typing again.

**Comment by iandunn** [comment]
After the modal is closed, it'd be cool if it inserted the failed word at the cursor, so you can try again from the beginning of the word, instead of just moving on to a new word.

Like if you accidentally pressed `k` instead of `m` in the word `jump`, it'd look like this:

`quick brown fox juk [pause with modal] jumped over...

**Comment by iandunn** [comment]
Another potential feature would be to have the modal show a list of common mistakes. Each one could link off to a webpage/video/whatever that discusses the problem and how to fix it. The user could check one of more of the problems to indicate which mistake they made. Over time, you could see which mistakes you make most often, and that would help you focus on correcting those.

**Comment by Miodec** [comment]
Have you tried the 'indicate typos' setting?

**Reply by iandunn** [reply]
I hadn't tried that before, I do like the `replace` setting, thanks!

It doesn't seem like it'd fulfill the intent of this feature, though. Were you thinking it does?
